### ISSUE TITLE
Allow extensions to choose to use whether to use CSS default data provider

### ISSUE THREAD

**Issue by braem** [microsoft/vscode#142946]
The CSS language service has had the parameter of whether to use the default data provider ([`useDefaultDataProvider`](https://github.dev/microsoft/vscode-css-languageservice/blob/c29c007451cd10d3c565250efd8aba44c54b2485/src/languageFacts/dataManager.ts#L36-L37)) for long while now and more recently [the exposing of `css.customData` for extensions](https://github.com/microsoft/vscode/issues/135459) which is lovely. However the bundled `css-language-features` extension [sets data providers with `useDefaultDataProvider` forced to `true`](https://github.dev/microsoft/vscode/blob/c64daad0ff2e7b5fd5e423e54af8a1da7aa2d712/extensions/css-language-features/server/src/cssServer.ts#L204).

Considering these go together in setting data providers, it would be fantastic to expose `useDefaultDataProvider` in the same way. This would make it very simple for extensions to support languages/frameworks that deviate from spec. Seems like it would be simple to do as well.

**Comment by aeschli** [comment]
Is the idea that extensions can contribute data providers that can indicate that they are a replacement for the default data provider? (= `useDefaultDataProvider` should be `false`).

**Comment by braem** [comment]
> Is the idea that extensions can contribute data providers that can indicate that they are a replacement for the default data provider? (= `useDefaultDataProvider` should be `false`).
> 
> 

Yes exactly! Though doing that with just customdata is quite limited; there is no way to specify values for completion, references, and other niceties available in the [languageservice's webCustomData](https://github.com/microsoft/vscode-css-languageservice/blob/main/src/data/webCustomData.ts).

For my use case I had to fork the languageservice and [essentially copy vscode's built-in css-language-features extension](https://github.com/braem/vscode-panorama-css) to connect to it. I also had to [make it its own language and disable all features from the built-in extension](https://github.com/braem/vscode-panorama-css/blob/main/package.json#L42-L164).

I'm not sure how common the use case is, but it would be amazing to be able to provide a replacement default data provider to avoid overriding the built-in extension entirely.

**Comment by aeschli** [comment]
If you want to work on a PR, that would be great. It would suggest adding it as a new property to the custom data file.
